rfd/0001-general-design.md

```diff
@@ -0,0 +1,433 @@
+         │ N │                                     │  F  │ ││   timeoutSec    ││      ┌─────┐      │  │  Pool   │ │
+         │ F │                                     │  R  │ │└─────────────────┘│      │  B  │      │  └─────────┘ │
+         │ I │                                     │  O  │ └───────────────────┘      │  A  │      │              │
+         │ G │       ┌─────────┐           ┌────┐  │  N  │                            │  C  │      │  ┌────────┐  │
+         │ U │     ┌─┤Request 1├─┐      ┌──┤ R1 ├─▶│  T  │                            │  K  │────┬─┼─▶│  addr1 │  │
+         │ R │     │ └─────────┘ │      │  └────┘  │  E  │                            │  E  │    │ │  └────────┘  │
+         │ E │     │             │      │          │  N  │                            │  N  │    │ │  ┌────────┐  │
+         └─┬─┘┌────────┐         ▼      │          │  D  ├────────────────────────────▶  D  │    ├─┼──▶  addr2 │  │
+           │  │        │     ┌──────┬──────┐       └─────┘                            └─────┘    │ │  └────────┘  │
+           └──│ Client │     │ 8090 │ mTLS │                                                     │ │  ┌────────┐  │
+              │        │     └──────┴──────┘       ┌─────┐                            ┌─────┐    └─┼─▶│  addr3 │  │
+              └────────┘         │      │          │  F  │                            │  B  │      │  └────────┘  │
+                   │             │      │          │  R  ├────────────────────────────▶  A  │      │  ┌────────┐  │
+                   │ ┌─────────┐ │      │          │  O  │                            │  C  │    ┌─┼─▶│  addr4 │  │
+                   └─┤Request 2├─┘      │  ┌────┐  │  N  │                            │  K  │────┤ │  └────────┘  │
+                     └─────────┘        └──┤ R2 ├─▶│  T  │      ┌─────────┐           │  E  │    │ │  ┌────────┐  │
+                                           └────┘  │  E  │ ┌────┤ options ├────┐      │  N  │    └─┼─▶│  addrN │  │
+                                                   │  N  │ │┌───┴─────────┴───┐│      │  D  │      │  └────────┘  │
+                                                   │  D  │ ││addr4,addr5,addrN││      └─────┘      │              │
+                                                   └────┬┴─│├─────────────────┤│──┐      ▲         └──────────────┘
+                                                        └──□│   timeoutSec    ││  │      │                         
+                                                           │└─────────────────┘│  └──────┘                         
+                                                           └───────────────────┘                                   
+```
+#### Frontend
+Is the part of the application customer can configure, and it represents Load Balancer
+object options.
+
+Frontend Available in Api Interface, and is entity in Storage Abstraction.
```

Comment:
What would you feel about cutting the scope and removing fronted and keeping configuration hardcoded  in code  and focusing in LB functionality ?